app/src/main/java/org/mozilla/fenix/compose/list/ExpandableListHeader.kt

```diff
@@ -0,0 +1,169 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.list
+
+import android.content.res.Configuration
+import androidx.annotation.DrawableRes
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Wrapper object for the ExpandableListHeader's state logic
+ *
+ * @param expanded Indicates whether the section of content is expanded.
+ * @param expandActionContentDescription The content description for expanding the section.
+ * @param collapseActionContentDescription The content description for collapsing the section.
+ */
+data class ExpandableListHeaderState(
+    val expanded: Boolean,
+    val expandActionContentDescription: String,
+    val collapseActionContentDescription: String,
+)
+
+/**
+ * Wrapper object for a sub action in an ExpandableListHeader
+ *
+ * @param iconDrawable Drawable used for the sub action.
+ * @param contentDescription Optional content description for the sub action.
+ * @param onClick Optional lambda for handling clicks.
+ */
+data class ExpandableListHeaderSubAction(
+    @DrawableRes val iconDrawable: Int,
+    val contentDescription: String?,
+    val onClick: () -> Unit = {}
+)
+
+/**
+ * Expandable header for sections of lists
+ *
+ * @param headerText The title of the header.
+ * @param state Indicates whether the section of content is expanded. If null, the Icon will be hidden.
+ * @param subAction Optional right-aligned element with an optional lambda.
+ * @param onClick Optional lambda for handling header clicks.
+ */
+@Composable
+fun ExpandableListHeader(
+    headerText: String,
+    state: ExpandableListHeaderState? = null,
```

Comment:
We discussed this over Slack, but my proposal here is to not nested state. Usually having nested state means we should just have a @Composable content parameter and allow components to be injected.